hubstorage/client.py

```diff
@@ -14,6 +17,24 @@
 __version__ = pkgutil.get_data('hubstorage', 'VERSION').strip()
 
 
+logger = logging.getLogger('HubstorageClient')
+
+_ERROR_CODES_TO_RETRY = (429, 503, 504)
+
+
+def _hc_retry_on_exception(err):
+    """Callback used by the client to restrict the retry to acceptable errors"""
+    if (isinstance(err, HTTPError) and err.response.status_code in _ERROR_CODES_TO_RETRY):
+        logger.warning("Server failed with %d status code, retrying (maybe)" % (err.response.status_code,))
+        return True
+
+    if (isinstance(err, ConnectionError) and err.args[0] == 'Connection aborted.' and
+            isinstance(err.args[1], BadStatusLine) and err.args[1][0] == repr('')):
+        logger.warning("Protocol failed with BadStatusLine, retrying (maybe)")
+        return True
```

Comment:
RemoteDisconnected inherits from BadStatusLine their should be no difference,
but actually, after testing the code in python3 it appears that it triggers a different structure for the same error. Catching it would look like:

```
                isinstance(err, ConnectionError)
                and isinstance(err.args[0], ProtocolError)
                and err.args[0].args[0] == 'Connection aborted.'
                and isinstance(err.args[0].args[1], BadStatusLine))
```

This is even terrible, I'm looking at an idiomatic way to check the list of chained exception, something like:

```if BadStatusLineError in all_chained_errors(err)```

Thanks for the note!